ruby/extractor/src/main.rs

```diff
@@ -170,20 +197,35 @@ fn main() -> std::io::Result<()> {
                                     Ok(str) => source = str.as_bytes().to_owned(),
                                     Err(msg) => {
                                         needs_conversion = false;
-                                        tracing::warn!(
-                                            "{}: character decoding failure: {} ({})",
-                                            &path.to_string_lossy(),
-                                            msg,
-                                            &encoding_name
+                                        diagnostics_writer.write(
+                                            &logger
```

Comment:
Good catch! This was just a silly copy/paste bug of mine. I added a `drop(logger)` statement right after the last use of `logger` to prevent me making the same mistake in the future.